### ISSUE TITLE
Mat-select accessibility behavior different when component is in a mat-dialog 

### ISSUE THREAD

**Issue by TQuinn33** [angular/components#21787]
When using VoiceOver for macOS, the mat-select options will be read when the mat-select has focus and the keyboard arrow keys are used to navigate through each option in the list. However, if the component is used in a mat-dialog, you can still navigate through the select option with the arrow keys when the select has focus, but the VoiceOver screenreader does not read the option values. 

This forces our users that rely on VoiceOver to interact differently with mat-selects when they are part of a mat-dialog. Is this expected or could this be fixed to behave the same as when mat-selects are not in a mat-dialog?

I have attached the code from the app folder in a small sample app created to reproduce the issue.
[app.zip](https://github.com/angular/components/files/5918558/app.zip)

**Comment by jelbourn** [comment]
Duplicate of #20900
